implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpMetricsSender.java

```diff
@@ -26,11 +26,14 @@
 public interface OtlpMetricsSender extends MetricsSender {
 
     /**
-     * Send a batch of OTLP Protobuf format metrics to a receiver.
+     * Send a batch of OTLP Protobuf format metrics to an OTLP receiver.
+     * @param address address of the OTLP receiver to which metrics will be sent
      * @param metricsData OTLP protobuf encoded batch of metrics
      * @param headers metadata to send as headers with the metrics data
+     * @throws Throwable when there is an exception in sending the metrics; the caller
+     * should handle this in some way such as logging the exception
      */
     @Override
-    void send(byte[] metricsData, Map<String, String> headers);
+    void send(String address, byte[] metricsData, Map<String, String> headers) throws Throwable;
```

Comment:
I think taking the address as a String is generic enough to work for other implementations than the HTTP one, but let me know if anyone has a concern about this. The implementation is responsible for parsing the address String. The OtlpMeterRegistry will always send the OtlpConfig#url as the address, so while it may semantically be weird if the address isn't actually a URL but the configured sender implementation can handle it (since it is a String that isn't necessarily a URL), this should work nonetheless.

Reply:
Thanks for the review. We could do something like that, and there's precedent in the codebase with HttpSender taking a Request object. I can take a closer look at it tomorrow.

Reply:
I'm totally fine calling it `address` but if you are interested in an alternative: `endpoint`.